### DISCUSSION TITLE
Parameters descriptions is shown only for the parent component

### DISCUSSION THREAD

**Discussion by daniele-zurico** [storybookjs/storybook#25191] [Help]
### Summary

Hi everyone my main problem is that the docgen is created only for one of the components and not for all of them. Here's an example: [Tabs](https://main--6069a6f47f4b9f002171f8e1.chromatic.com/?path=/docs/dcxlibrary-tabs-documentation--docs), as you can see here we got all the available properties only for the [TabGroup component](https://github.com/Capgemini/dcx-react-library/blob/main/src/tabGroup/TabGroup.tsx) but we don't have the properties related to [Tab](https://github.com/Capgemini/dcx-react-library/blob/main/src/tabGroup/components/Tab.tsx). There's any chance that I can include them?

<img width="1145" alt="Screenshot 2023-12-12 at 11 49 13" src="https://github.com/storybookjs/storybook/assets/3193095/3e427077-ef8b-4909-8c6b-53a4d2b69572">

### Additional information

This is how the [Documentation](https://github.com/Capgemini/dcx-react-library/blob/main/stories/TabGroup/Documentation.mdx) has been created for the following component

**Comment by arcticShadow** [comment]
Hi @daniele-zurico 

Disclaimer: I am not associated with Storybook - just a user.

Based on your description, it sounds like you are expecting multiple components to be documented together - To the best of my knowledge, Storybook autodocs will only document a single Component at one time. 

The component that is documented, is determined by the component that you pass to the meta block i.e. for your use case

```typescript
<Meta component={TabGroup} />
```

If you wanted `<Tab />` to be documented as well, this needs to be on on a separate page. 

If you have the option of changing your code, you may consider to pass an array of `tabs` , where each item in the array is an object, typed with the prop types of your current tab component. The TabGroup then maps over the tabs array, and returns Tabs with the props applied.

**Reply by daniele-zurico** [reply]
Hi @arcticShadow thanks for your reply however I'm not sure you got the real problem. When you create a component sometimes you have a child component that is part of it. In my example you have something like this:
```jsx
<TabGroup propA="something">
   <Tabs propB="else">
<TabGroup/>
```
However when the plugin generate the list of available properties you'll find only PropA and you'll never see PropB.

**Reply by arcticShadow** [reply]
If you included an Example/code there, it didn't show up :-)

Possibly your example further clarifies your issue, but the below may or may not help :-)

I can see from your original example what you are including `<Tab>` as a child of `<TabGroup>`  - There is nothing wrong with this approach it is compliant and will work. 

Storybook however treats the two components as independent components - because there is nothing that explicitly ties them together - they are just two react components.  i.e. Its still perfectly compliant with react for you to take your `Tab` component, and add it directly to a `<div>` in JSX. It probably wouldn't work the way you expect, but it is compliant to React.

**Reply by daniele-zurico** [reply]
Unfortunately, I disagree with your point of view. Tab do now exists without TabGroup in fact if you try so it will throw an error given the [context](https://github.com/Capgemini/dcx-react-library/blob/main/src/tabGroup/TabGroup.tsx#L193) not allow it.
On the other side the problem is not if you can use or not use the component on his own but have a way to document all the properties of Tabs as well when you document the full [Component](https://main--6069a6f47f4b9f002171f8e1.chromatic.com/?path=/docs/dcxlibrary-tabs-documentation--docs) otherwise the user will not be able to understand what is available to use it
